tensorflow_addons/optimizers/momentum_lars.py

```diff
@@ -0,0 +1,206 @@
+    tensors use the default momentum optimizer.
+    """
+
+    @typechecked
+    def __init__(
+        self,
+        learning_rate: float = 0.001,
+        momentum: float = 0.9,
+        weight_decay: float = 0.0001,
+        # The LARS coefficient is a hyperparameter
+        eeta: float = 0.001,
+        epsilon: float = 0.0,
+        name: str = "MomentumLARS",
+        # Enable skipping variables from LARS scaling.
+        skip_list: Optional[str] = (),
+        use_nesterov: bool = False,
+        clip: bool = False,
+        **kwargs
+    ):
+        """Construct a new LARS Optimizer.
+
+      Args:
+          learning_rate: A `Tensor` or floating point value. The base learning rate.
+          momentum: A floating point value. Momentum hyperparameter.
+          weight_decay: A floating point value. Weight decay hyperparameter. Adding an
+             L2 regularizer to a Keras variable is not equivalent to how the LARS paper
+             handles weight decay. In the LARS paper, when computing the "trust"
+             coefficient, the magnitude of the gradient and the magnitude weights
+             are added together. But if an L2 regularizer is added to a Keras variable,
+             the gradient and weights are first added together, and then the magnitude is taken
+          eeta: LARS coefficient as used in the paper. Default set to LARS
+            coefficient from the paper. (eeta / weight_decay) determines the highest
+            scaling factor in LARS.
+          epsilon: Optional epsilon parameter to be set in models that have very
+          small gradients. Default set to 0.0.
+          name: Optional name prefix for variables and ops created by LARS Optimizer.
+          skip_list: List of strings to enable skipping variables from LARS scaling.
+            If any of the strings in skip_list is a subset of var.name, variable
+            'var' is skipped from LARS scaling. For a typical classification model
+             with batch normalization, the skip_list is ['batch_normalization',
+             'bias']
+          use_nesterov: when set to True, nesterov momentum will be enabled
+          clip: when set to True, learning rate clipping will be enabled.
+
+      Raises:
+          ValueError: If a hyperparameter is set to a non-sensical value.
+      """
+        if momentum < 0.0:
+            raise ValueError("momentum should be positive: %s" % momentum)
+        if weight_decay < 0.0:
+            raise ValueError("weight_decay should be positive: %s" % weight_decay)
+        super(MomentumLARS, self).__init__(name=name, **kwargs)
+
+        self._skip_list = skip_list
+        self._set_hyper("learning_rate", learning_rate)
+        self._set_hyper("momentum", momentum)
+        self._set_hyper("weight_decay", weight_decay)
+        self._set_hyper("eeta", eeta)
+        self._set_hyper("epsilon", epsilon)
+        self._set_hyper("use_nesterov", use_nesterov)
+        self._set_hyper("use_clipping", clip)
```

Comment:
For arguments with `bool` type, keras semantics is to set it as member directly, like
```
self.use_nesterov = use_nesterov
self.use_clipping = use_clipping
```
and access it without casting to bool.

https://github.com/tensorflow/tensorflow/blob/r2.1/tensorflow/python/keras/optimizer_v2/adam.py#L143